### PR TITLE
Publish block V1 flag fix

### PR DESCRIPTION
**Pull request by avalonche**
## 📝 Summary

Same as #582 but with some linter fixes and comments addressed as for some reason the linter was panicking

## ⛱ Motivation and Context

See #581

## 📚 References

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test-race`
* [x] `go mod tidy`
* [x] I have seen and agree to `CONTRIBUTING.md`

### CODE REVIEW

**Comment by avalonche** [comment]
for some reason the linter was panicking in #582, this includes a linter upgrade

File: .golangci.yaml [hunk 1]
```diff
@@ -30,6 +30,7 @@ linters:
     - deadcode
     - varcheck
     - interfacebloat
+    - exhaustruct
```
Comment (metachris) [inline .golangci.yaml:33]: why disable this linter rule?

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: ^1.20
+          go-version: ^1.21
```
Comment (metachris) [inline .github/workflows/checks.yml:17]: could use go 1.22

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/flashbots/mev-boost-relay
 
-go 1.20
+go 1.21
```
Comment (metachris) [inline go.mod:3]: could use go 1.22

**Review by metachris — APPROVED** [review]
lgtm

File: .golangci.yaml (hunk 1 above)
Comment (avalonche) [reply .golangci.yaml:33]: it was causing too many linter issues as there are many paths that are not ignored. I will have a follow up to properly update the go version and linter

**Pull request merged**